src/main/cms/cms_menu_imu.c

```diff
@@ -480,12 +480,75 @@ CMS_Menu cmsx_menuCopyProfile = {
 
 #endif
 
+static uint8_t gyroConfig_gyro_kalman_enable;
+static uint16_t gyroConfig_gyro_kalman_q;
+static uint16_t gyroConfig_gyro_kalman_r;
+static uint16_t gyroConfig_gyro_kalman_p;
+
+static long cmsx_kalmanOnEnter(void)
+{
+    gyroConfig_gyro_kalman_enable =  gyroConfig()->gyro_kalman_enable;
+    gyroConfig_gyro_kalman_q =  gyroConfig()->gyro_kalman_q;
+    gyroConfig_gyro_kalman_r = gyroConfig()->gyro_kalman_r;
+    gyroConfig_gyro_kalman_p = gyroConfig()->gyro_kalman_p;
+
+    return cmsx_RateProfileOnEnter();
+}
+
+static long cmsx_kalmanWriteback(const OSD_Entry *self)
+{
+    gyroConfigMutable()->gyro_kalman_enable =  gyroConfig_gyro_kalman_enable;
+    gyroConfigMutable()->gyro_kalman_q =  gyroConfig_gyro_kalman_q;
+    gyroConfigMutable()->gyro_kalman_r =  gyroConfig_gyro_kalman_r;
+    gyroConfigMutable()->gyro_kalman_p =  gyroConfig_gyro_kalman_p;
+
+    return cmsx_RateProfileWriteback(self);
+}
+
+static OSD_Entry cmsx_menuKalmanEntries[] =
+{
+    { "-- KALMAN --",    OME_Label,  NULL, rateProfileIndexString, 0},
+    { "ENABLED",         OME_Bool,   NULL, &gyroConfig_gyro_kalman_enable, 0 },
+
+    { "KALMAN Q",        OME_UINT16, NULL, &(OSD_UINT16_t) { &gyroConfig_gyro_kalman_q, 0, 16000, 10 }, 0 },
+    { "KALMAN R",        OME_UINT16, NULL, &(OSD_UINT16_t) { &gyroConfig_gyro_kalman_r, 0, 16000, 10 }, 0 },
+    { "KALMAN P",        OME_UINT16, NULL, &(OSD_UINT16_t) { &gyroConfig_gyro_kalman_p, 0, 16000, 10 }, 0 },
+
+    { "RF RATE ENABLED", OME_Bool,   NULL, &rateProfile.rfRatesEnabled, 0 },
+
+    { "RF RATE ROLL",    OME_UINT16, NULL, &(OSD_UINT16_t) { &rateProfile.rfRate[FD_ROLL], 200, 1500, 10 }, 0 },
```

Comment:
Step for Rate should be 5, all can probably be 5 but rate should be 1 or 5 not 10.